payjoin-cli/src/app/rpc.rs

```diff
@@ -81,17 +81,18 @@ impl AsyncBitcoinRpc {
             .json(&request_body)
             .basic_auth(&self.username, Some(&self.password));
 
-        let response = request.send().await.context("Failed to send RPC request")?;
+        let response =
+            request.send().await.with_context(|| format!("RPC '{}': connection failed", method))?;
 
-        if !response.status().is_success() {
-            return Err(anyhow!("RPC request failed with status: {}", response.status()));
-        }
-
-        let json: RpcResponse<T> = response.json().await.context("Failed to parse RPC response")?;
+        let json = response
+            .json::<RpcResponse<T>>()
```

Comment:
nit: :This was `.json()` without the generic type definition before and that worked just fine